api/src/main/java/org/apache/iceberg/Schema.java

```diff
@@ -331,6 +363,16 @@ public Schema caseInsensitiveSelect(Collection<String> names) {
     return internalSelect(names, false);
   }
 
+  /**
+   * Checks whether this schema is equivalent to another schema while ignoring the schema ID.
+   * @param anotherSchema another schema
+   * @return true if this schema is equivalent to the given schema
+   */
+  public boolean sameSchema(Schema anotherSchema) {
+    return asStruct().equals(anotherSchema.asStruct()) &&
+        identifierFieldIds().equals(anotherSchema.identifierFieldIds());
```

Comment:
Using the `Set` to check the equality should be correct,  I was thinking that we could use the simple `Arrays.equals(identifierFieldIds, anotherSchema.identifierFieldIds)`, that means the identifier field id's order will matter and that's not what we expected.

Reply:
Yeah, I think it should be fine to check the sets.